lib/src/render/utils/render_box_layout.dart

```diff
@@ -0,0 +1,21 @@
+import 'dart:ui';
+
+import 'package:flutter/rendering.dart';
+
+extension RenderBoxLayout on RenderBox {
+  /// Returns size of render box based on provided [BoxConstraints].
+  ///
+  /// `dry` flag indicates whether a real layout pass should be
+  /// executed on render box. Default value assumes that we don't
+  /// want to perform real layout and only get potential size of box.
+  Size getLayoutSize(BoxConstraints constraints, {bool dry = true}) {
+    final Size childSize;
+    if (dry) {
+      childSize = this.getDryLayout(constraints);
+    } else {
+      this.layout(constraints, parentUsesSize: true);
+      childSize = this.size;
+    }
+    return childSize;
+  }
```

Comment:
I do not think I like this as an extension because it is not clear that this should be used on a child.

Instead, I would write it as:

```suggestion
/// Returns the size of a child render box given the provided [BoxConstraints].
///
/// The `dry` flag indicates that no real layout pass but only a dry
/// layout pass should be executed on the render box.
/// Defaults to true.
Size getLayoutSize(BoxConstraints constraints, {bool dry = true}) {
  final Size childSize;
  if (dry) {
    childSize = this.getDryLayout(constraints);
  } else {
    this.layout(constraints, parentUsesSize: true);
    childSize = this.size;
  }
  return childSize;
}
```